packages/node-type-resolver/src/NodeTypeCorrector/PregMatchTypeCorrector.php

```diff
@@ -108,7 +108,10 @@ private function getVariableUsages(Variable $variable): array
         }
 
         return $this->betterNodeFinder->find((array) $scope->stmts, function (Node $node) use ($variable): bool {
-            return $node instanceof Variable && $node->name === $variable->name;
+            if (! $node instanceof Variable) {
+                return false;
+            }
+            return (bool) ($node->name === $variable->name);
```

Comment:
This also returns bool, so no type cast is needed

```suggestion
            return $node->name === $variable->name;
```

Reply:
implemented.